salttesting/jenkins.py

```diff
@@ -941,12 +1044,222 @@ def delete_parallels_vm(options):
+               options.vm_name, 'test.ping']
+
+        # Attempt to connect to the new minion, it can take a while with a new
+        # install
+        retries = 0
+        while retries <= 12:
+
+            retries += 1
+            stdout, stderr, exitcode = run_command(
+                cmd, options, return_output=True, stream_stdout=False,
+                stream_stderr=False)
+            if exitcode:
+                print_bulleted(
+                    options,
+                    'Failed to return a ping from the minion. Exit code: {0}'
+                    ''.format(exitcode), 'RED'
+                )
+                if retries >= 12:
+                    sys.exit(exitcode)
+
+            if not stdout.strip():
+                print_bulleted(
+                    options,
+                    'Failed to return a ping from the minion (no output).',
+                    'RED')
+                if retries >= 12:
+                    sys.exit(1)
+
+            try:
+                # Load the return with JSON
+                ping = json.loads(stdout.strip())
+                print_bulleted(options, 'Loaded JSON: {0}'.format(ping))
+            except (ValueError, TypeError):
+                print_bulleted(
+                    options,
+                    'Failed to load any JSON from {0!r}'.format(stdout.strip()),
+                    'RED')
+
+            if ping[options.vm_name] is True:
+
+                # Returned ping, reboot
+                cmd = ['salt', '--out=json', '-l', options.log_level,
+                       options.vm_name, 'system.reboot', '0', 'True']
+
+                stdout, stderr, exitcode = run_command(
+                    cmd, options, return_output=True,
+                    stream_stdout=False, stream_stderr=False)
+                if exitcode:
+                    print_bulleted(
+                        options,
+                        'Failed to reboot the minion. Exit code: {0}'
+                        ''.format(exitcode), 'RED'
+                    )
+                    if retries >= 12:
+                        sys.exit(1)
+
+                if not stdout.strip():
+                    print_bulleted(
+                        options,
+                        'Failed to reboot the minion (no output).',
+                        'RED')
+                    if retries >= 12:
+                        sys.exit(1)
+
+                try:
+                    # Load the return
+                    res = json.loads(stdout.strip())
+                    print_bulleted(options, 'Loaded JSON: {0}'.format(res))
+                except (ValueError, TypeError):
+                    print_bulleted(
+                        options,
+                        'Failed to load any JSON from {0!r}'
+                        ''.format(stdout.strip()),
+                        'RED')
+
+                # It should return True
+                if res[options.vm_name] is True:
```

Comment:
Same problem here as above. You will not have `res` defined if you hit the exception handler above.